server/stream.go

```diff
@@ -2287,6 +2353,21 @@ func (mset *stream) internalSendLoop() {
 			if pm.o != nil && pm.seq > 0 && !didDeliver {
 				pm.o.didNotDeliver(pm.seq)
 			}
+		case <-mch:
+			for im := mset.pending(); im != nil; {
+				// If we are clustered we need to propose this message to the underlying raft group.
+				if isClustered {
+					mset.processClusteredInboundMsg(im.subj, im.rply, im.hdr, im.msg)
+				} else {
+					mset.processJetStreamMsg(im.subj, im.rply, im.hdr, im.msg, 0, 0)
+				}
+				// Do this here to nil out below vs up in for loop.
+				next := im.next
+				im.next, im.hdr, im.msg = nil, nil, nil
+				if im = next; im == nil {
+					im = mset.pending()
```

Comment:
Same comment than before, and even more important here since you would possibly starve the other channel you are receiving on.

Reply:
Will remove this one.